### PR TITLE
transmission: update to 3.00

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.4 19E287
Xcode 11.5 11E608c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: net/transmission/Portfile [hunk 1]
```diff
@@ -69,21 +56,19 @@ destroot {
...
+        ${destroot}${prefix}/share/man/man1
+        
+        xinstall -m 644 -W ${worksrcpath}/utils \
+        transmission-remote.1 \
         ${destroot}${prefix}/share/man/man1
    }
 }
 
-if {([vercmp $xcodeversion 7.0] < 0) || ([vercmp ${macosx_deployment_target} 10.9] < 0)} {
+if {([vercmp $xcodeversion 8.0] < 0) || ([vercmp ${macosx_deployment_target} 10.10] < 0)} {
```
Comment (cjones051073) [inline net/transmission/Portfile:68]: Please fix the warning message below to match the changed versions.
Ideally make them local variables, so you do not have to hardcode the same versions in two places.

File: net/transmission/Portfile [hunk 2]
```diff
@@ -4,10 +4,11 @@ PortSystem      1.0
 PortGroup       github 1.0
 PortGroup       xcode 1.0
 
-github.setup    transmission transmission 2.94
-revision        1
+github.setup    transmission transmission 3.00
+set git_commit  bb6b5a062ee594dfd4b7a12a6b6e860c43849bfd
```
Comment (cjones051073) [inline net/transmission/Portfile:8]: Why are you using a explicit commit ? Please change this to use the 3.0.0 release tag.

**Comment by cjones051073** [comment]
Have you discussed with the current maintainer adding yourself as a new maintainer?

**Comment by i0ntempest** [comment]
Not yet, the last time I updated this he replied pretty quick so I assume he'll look at this soon. Or it's better to do this in another PR?

**Comment by cjones051073** [comment]
> Not yet, the last time I updated this he replied pretty quick so I assume he'll look at this soon. Or it's better to do this in another PR?

I would be tempted to remove it here and submit a second PR for it. As it stands without that change I think these updates can be merged as soon as the tests complete.

**Comment by kurthindenburg** [comment]
I'm a bit concerned about all the lines being removed.  Also why is the pre-fetch git needed?

**Comment by i0ntempest** [comment]
Afaik the patch and reinplaces to the Xcode project file are no longer needed, if not removed they’ll actually break the build. The reason is prolly that some dependencies are now submodules so we don’t need to mess with the path anymore. This is also why that git command is needed (to download all submodules). Upstream developer will provide a complete tarball for the next version so we can change it back.

**Comment by i0ntempest** [comment]
Actually it seems that all dependencies can be removed because they're all provided by submodules.

File: net/transmission/Portfile [hunk 3]
```diff
@@ -43,22 +44,6 @@ variant daemon description {Builds headless daemon} {
...
-}
-
-post-patch {
-    reinplace -E "s|@@PREFIX@@|${prefix}|g" \
-        ${worksrcpath}/Transmission.xcodeproj/project.pbxproj
-}
-
-pre-build {
-    if {${os.platform} eq "darwin" && ${os.version} >= 18} {
-        system -W ${worksrcpath}/macosx "/usr/bin/codesign --sign - Sparkle.framework"
```
[the remark this answers is not in the document]
Comment (i0ntempest) [reply net/transmission/Portfile:58]: Code signing is now handled correctly during the build. The app built is already signed even without this.

**Comment by i0ntempest** [comment]
It might be possible by patching the xcode project file to change the paths. But this could be really hard to do and fragile at best

**Comment by i0ntempest** [comment]
Also not all submodules are in macports (libnatpmp).

**Comment by cjones051073** [comment]
> Also not all submodules are in macports (libnatpmp).

well that could be fixed, by providing new ports for them.

I wasn't suggesting hacking the build system, more if they provide an explicit way to set the path to externals, and thus effectively disable the use of their own versions. Many packages provide hooks to allow this.

**Comment by i0ntempest** [comment]
In 2.94 the build is already hacked (see pre-patch and post-patch). Xcode projects seem to hardcode dependency paths. Not sure if there's a better way, I'm looking into it

**Comment by i0ntempest** [comment]
Okay I've added back and tweaked the build hack to make it uses some of the macports packages. I think it's possible to continue this to the point that submodules are no longer needed (all provided by macports), but it'll need some new ports and huge amounts of trail and error (at least for someone like me).

**Comment by i0ntempest** [comment]
Also since the upstream developer will include all submodules in the next release, to continue with current state, those submodules replaced by mp will prolly need to be deleted during build to avoid redefinition errors.

**Comment by kurthindenburg** [comment]
It builds here on Catalina but won't start

  Library not loaded: @rpath/Sparkle.framework/Versions/A/Sparkle
  Referenced from: /Applications/MacPorts/Transmission.app/Contents/MacOS/Transmission
  Reason: no suitable image found.  Did find:
	/Applications/MacPorts/Transmission.app/Contents/MacOS/../Frameworks/Sparkle.framework/Versions/A/Sparkle: code signature in (/Applications/MacPorts/Transmission.app/Contents/MacOS/../Frameworks/Sparkle.framework/Versions/A/Sparkle) not valid for use in process using Library Validation: mapped file has no Team ID and is not a platform binary (signed with custom identity or adhoc?)
	/Applications/MacPorts/Transmission.app/Contents/MacOS/../Frameworks/Sparkle.framework/Versions/A/Sparkle: stat() failed with errno=1

**Comment by i0ntempest** [comment]
Can you try adhoc sign that framework bundle? That might allow it to start.

**Comment by kurthindenburg** [comment]
sudo /usr/bin/codesign --sign -   /Applications/MacPorts/Transmission.app/Contents/Frameworks/Sparkle.framework

/Applications/MacPorts/Transmission.app/Contents/Frameworks/Sparkle.framework: is already signed

**Comment by i0ntempest** [comment]
Force it with --force, if that doesn’t work try signing the whole app with --force and --deep

**Comment by kurthindenburg** [comment]
/usr/bin/codesign --force --deep --sign  -   /Applications/MacPorts/Transmission.app/Contents/Frameworks/Sparkle.framework

/Applications/MacPorts/Transmission.app/Contents/Frameworks/Sparkle.framework: replacing existing signature

/Applications/MacPorts/Transmission.app/Contents/Frameworks/Sparkle.framework: bundle format is ambiguous (could be app or framework)

**Comment by i0ntempest** [comment]
Welp, I’ll investigate later

**Comment by i0ntempest** [comment]
Figured it out, 3.00 tarball is so problematic. What happened is during extraction all symlinks inside the framework are followed and become the actual files/folders, and codesign program doesn't like that. Extracting the tarball with The Unarchiver app also produces errors like "Could not create symlink".

**Comment by kurthindenburg** [comment]
Ok thanks - it build and runs here -  I haven't fully tested it by using files yet

**Comment by kurthindenburg** [comment]
It works using a couple of files I tested with.  I'm fine w/ you committing this.

**Comment by kurthindenburg** [comment]
Thanks for spending so much time on this - you can add yourself as maintainer

**Pull request merged**